web/packages/teleterm/src/ui/services/keyboardShortcuts/keyboardShortcutsService.ts

```diff
@@ -89,14 +117,24 @@ export class KeyboardShortcutsService {
   private getShortcutAction(
     event: KeyboardEvent
   ): KeyboardShortcutAction | undefined {
-    const getEventKey = () =>
-      event.key.length === 1 ? event.key.toUpperCase() : event.key;
-
-    const accelerator = [...this.getPlatformModifierKeys(event), getEventKey()]
+    // skip modifier-only events
+    if (
+      event.code.includes('Shift') ||
+      event.code.includes('Meta') ||
+      event.code.includes('Alt') ||
+      event.code.includes('Control')
+    ) {
+      return;
+    }
+    const accelerator = [
+      ...this.getPlatformModifierKeys(event),
+      getKeyCode(event),
+    ]
       .filter(Boolean)
       .join('+');
 
-    return this.acceleratorsToActions.get(accelerator);
+    // always return the first action (in case of duplicated accelerators)
+    return this.acceleratorsToActions.get(accelerator)?.[0];
```

Comment:
(attaching a comment to a not-so relevant place in the code to avoid creating a top-level comment)

One thing I forgot to mention, the shift symbol looks totally fine in the tooltip but on the page with the list of shortcuts it looks terrible. 🫠

![shift symbol](https://user-images.githubusercontent.com/27113/221221600-d666e103-de28-45d0-9318-8660e461f8bc.png)

Symbols in general look terrible in the monospace font, I guess one way to address it would be to display symbol characters in a non-monospace font and the rest of characters in a monospace font (so the user can see a difference between `0` and `O`).

But that's too much for now I think, we can come back to this later.

Reply:
Uh, this doesn't look good, but I agree, we can come back later to it.